### PR TITLE
Show progress indication in admin needle table

### PR DESCRIPTION
**Pull request by Martchus**
- Improve situation of unresponsive needle table:
  https://progress.opensuse.org/issues/13196
- Next step is implementing server side paging.
- Also fix indentation in admin_needle.js

### CODE REVIEW

**Comment by Martchus** [comment]
BTW: Using server side paging improves the performance significantly. However, what I've got so far is still buggy so I'm just proposing this commit for now.

**Comment by okurz** [comment]
The only change I can see is "fixing indentation". Maybe the other change is hidden in there. I suggest to not mix simple coding style changes with feature changes.

**Comment by coolo** [comment]
the actual change is adding "processing": true,

Having an extra commit for that sounds a bit overkill, but I agree it's hard to spot :)

File: assets/javascripts/admin_needle.js [hunk 1]
```diff
@@ -1,88 +1,88 @@
...
-		     if (type === 'display' && data != 'never') {
+        { "ajax": ajaxUrl(),
+            deferRender: true,
+            "columns": [
+            { "data": "directory" },
+            { "data": "filename" },
+            { "data": "last_seen" },
+            { "data": "last_match" }
+            ],
+            "processing": true,
```
Comment (Martchus) [inline assets/javascripts/admin_needle.js]: Here's the actual change - this will enable the progress indication provided by DataTable.

**Comment by Martchus** [comment]
Server-side paging should work now.

If the filter for limiting match and/or use are enabled the query is still slow, though.

File: lib/OpenQA/WebAPI/Controller/Admin/Needle.pm [hunk 2]
```diff
@@ -47,12 +47,43 @@ sub ajax {
...
+    $params->{join} = [qw/directory last_seen last_match/];
+
+    # Parameter for order
+    my @columns = qw(directory.name filename last_seen.t_created last_match.t_created);
+    my @order_by_params;
+    my $index = 0;
+    while (1) {
+        my $column_index = $self->param("order[$index][column]");
+        my $column_order = $self->param("order[$index][dir]");
+        last
```
Comment (coolo) [inline lib/OpenQA/WebAPI/Controller/Admin/Needle.pm]: please just forget that perl has post conditions. If you need 4 lines to break the if - a post condition is just not the right thing.

**Comment by coolo** [comment]
most of the time seems to be spent in this query - possibly worth an index:

Took 2.16748190 seconds executed: SELECT MIN( me.id ) FROM job_modules me WHERE ( t_created >= ? ): '2016-09-06 15:37:13'.

**Comment by Martchus** [comment]
Not sure whether an index would improve this because it is a '>=' condition and the minimum is computed.

I tried to get rid of the extra queries for last seen/match because I now have to join last_seen and last_match anyways for the ordering. However, I was not able to use the joined columns in the condition.

**Comment by coolo** [comment]
trying this now - I think the processing part is now painful. it's just blinking for microseconds :)

**Pull request merged**